boot/boot.go

```diff
@@ -91,7 +91,12 @@ type Device interface {
 // always be NOP).
 func Participant(s snap.PlaceInfo, t snap.Type, dev Device) BootParticipant {
 	if applicable(s, t, dev) {
-		return &coreBootParticipant{s: s, t: t}
+		bs, err := bootStateFor(t, dev)
+		if err != nil {
+			// all internal errors at this point
+			panic(err)
```

Comment:
I'm a bit uneasy about this panic, I know it's internal but I wonder if a bug in bootStateFor or something unexpected on the system may put us into a situation where we try to run a task like refresh, try to update the boot, panic, restart and panic again, i.e. into a situation where we cannot recover from via a refresh. Or am I too paranoid here?

Reply:
the invariant here is that for any t and dev for which applicable is true bootStateFor should not error, this should be well covered between unit tests and spread already.